### PR TITLE
Refactor getting-started-reactive guide according to new resteasy-reactive extension codestart

### PR DESCRIPTION
**Pull request by aureamunoz**
Related to #13793

Could you please take a look @geoand @ia3andy ?

### CODE REVIEW

File: docs/src/main/asciidoc/getting-started-reactive.adoc [hunk 1]
```diff
@@ -171,7 +171,7 @@ mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
     -DprojectArtifactId=getting-started-reactive \
     -DclassName="org.acme.quickstart.ReactiveGreetingResource" \
     -Dpath="/hello" \
-    -Dextensions="resteasy-mutiny, resteasy-jackson"
+    -Dextensions="resteasy-reactive, rest-client"
```
Comment (geoand) [inline docs/src/main/asciidoc/getting-started-reactive.adoc:174]: Why the rest-client?

It is not needed AFAIK

Comment (aureamunoz) [reply docs/src/main/asciidoc/getting-started-reactive.adoc:174]: I need it for using `@@SseElementType` [here](https://github.com/aureamunoz/quarkus-quickstarts/blob/6ab7eb483f4f0931e6d609f461ea880df3c777d8/getting-started-reactive/src/main/java/org/acme/getting/started/ReactiveGreetingResource.java#L37)

Comment (geoand) [reply docs/src/main/asciidoc/getting-started-reactive.adoc:174]: OK, that is wrong.

You should use `@org.jboss.resteasy.reactive.RestSseElementType` instead and remove the rest-client dependency

Comment (aureamunoz) [reply docs/src/main/asciidoc/getting-started-reactive.adoc:174]: OK. I didn't know it. Thank you

**Review by geoand — CHANGES_REQUESTED** [review]
See https://github.com/quarkusio/quarkus/pull/14009/files#r546823555

File: docs/src/main/asciidoc/getting-started-reactive.adoc (hunk 1 above)
Comment (aureamunoz) [reply docs/src/main/asciidoc/getting-started-reactive.adoc:174]: org.jboss.resteasy.annotations.jaxrs.PathParam isn't in `quarkus-resteasy-reactive` dependency, is it expected?

Comment (geoand) [reply docs/src/main/asciidoc/getting-started-reactive.adoc:174]: Yes, it is expected. You can omit the annotation entirely and the code should work properly

**Comment by aureamunoz** [comment]
It should be ok now :-)

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by geoand** [comment]
Do you also have a PR for the quickstart itself?

**Comment by aureamunoz** [comment]
yep https://github.com/quarkusio/quarkus-quickstarts/pull/734

**Comment by geoand** [comment]
Ah darn... I missed the fact that there where 2 commits that should have been squashed :(

**Comment by aureamunoz** [comment]
And I think there is a minor bug in the codestart.yml, the guide is not correct (It should be https://quarkus.io/guides/getting-started-reactive) and [it's rest-json](https://github.com/quarkusio/quarkus/blob/cba756bf5303a8e8b7fa9d87715e2a2cb7134cf9/devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/examples/resteasy-reactive-example/codestart.yml#L15)

**Comment by geoand** [comment]
OK, please open a new PR

**Comment by aureamunoz** [comment]
yes, will do soon, I'm fixing another thing :-)

**Comment by aureamunoz** [comment]
new PR done, I have pinged you in it.

**Comment by ia3andy** [comment]
Nice!!

**Pull request merged**